crates/runc-shim/src/service.rs

```diff
@@ -78,8 +78,19 @@ impl Shim for Service {
             }
             None => {}
         }
+        #[cfg(not(target_os = "linux"))]
+        let thp_disabled = String::new();
+        #[cfg(target_os = "linux")]
+        let thp_disabled = match prctl::get_thp_disable() {
+            Ok(x) => {
+                let _ = prctl::set_thp_disable(true);
+                x.to_string()
```

Comment:
is it possible that you get `Ok(false)` and the set it to true and this will return false?  The does weren't clear (https://docs.rs/prctl/latest/prctl/fn.get_thp_disable.html)

Reply:
Our goal is to ```set thp disabled = true``` on the shim side and then restore thp disabled before starting runc.
So we only need to focus on the return value of the function ```get_thp_disabled```, which is ```Result<bool, i32>```.
The return value of the function ```set_thp_disabled``` is ```Result<(), i32>```, we don't care if the setting is successful, because even if the setting failed, we should not exit the shim process, therefore, there is no need to pay attention to the ```set_thp_disabled``` function's return value.

Reply:
> The return value of the function `set_thp_disabled` is `Result<(), i32>`, we don't care if the setting is successful, because even if the setting failed, we should not exit the shim process, therefore, there is no need to pay attention to the `set_thp_disabled` function's return value.

could you add a comment in the code that indicates this? I worry about someone doing maitaince long term and wondering why return value and failure case is ignored.

Reply:
Thanks for your suggestion, done.